nix/image.go

```diff
@@ -158,19 +173,9 @@ func NewImageFromDir(directory string) (image types.Image, err error) {
 			Digest:    l.Digest.String(),
 			DiffIDs:   v1ImageConfig.RootFS.DiffIDs[i].String(),
 		}
-		switch l.MediaType {
-		case "application/vnd.docker.image.rootfs.diff.tar":
-			layer.MediaType = v1.MediaTypeImageLayer
-		case "application/vnd.docker.image.rootfs.diff.tar.gzip":
-			layer.MediaType = v1.MediaTypeImageLayerGzip
-		case "application/vnd.oci.image.layer.v1.tar":
-			layer.MediaType = l.MediaType
-		case "application/vnd.oci.image.layer.v1.tar+gzip":
-			layer.MediaType = l.MediaType
-		case "application/vnd.oci.image.layer.v1.tar+zstd":
-			layer.MediaType = l.MediaType
-		default:
-			return image, fmt.Errorf("Unsupported media type: %q", l.MediaType)
+		err = layer.SetMediaTypeFromDescriptor(l)
+		if err != nil {
+			return image, err
```

Comment:
Not required for this PR, I just thought at first I needed this code a third time which turned out not to be the case. But I left that refactoring in anyways (in a separate commit). I can remove or change it.

Reply:
Fine for me.